DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/ManageCourseController.cs

```diff
@@ -99,5 +114,98 @@ EditLearningPathwayDefaultsViewModel model
 
             return RedirectToAction("Index", new { customisationId = model.CustomisationId });
         }
+
+        [HttpGet]
+        [ServiceFilter(typeof(RedirectEmptySessionData<EditLearningPathwayDefaultsData>))]
+        [Route("AutoRefreshOptions")]
+        public IActionResult EditAutoRefreshOptions(int customisationId)
+        {
+            var centreId = User.GetCentreId();
+            var categoryId = User.GetAdminCategoryId()!;
+
+            var courseDetails = courseService.GetCourseDetailsForAdminCategoryId(
+                customisationId,
+                centreId,
+                categoryId.Value
+            );
+
+            var model = new EditAutoRefreshOptionsViewModel(courseDetails!);
+
+            SetViewBagCourseOptions(customisationId, model.RefreshToCustomisationId);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [ServiceFilter(typeof(RedirectEmptySessionData<EditLearningPathwayDefaultsData>))]
+        [Route("AutoRefreshOptions")]
+        public IActionResult SaveAutoRefreshOptions(
+            int customisationId,
+            EditAutoRefreshOptionsViewModel model
+        )
+        {
+            if (customisationId != model.CustomisationId)
+            {
+                return new StatusCodeResult(500);
+            }
+
+            if (!ModelState.IsValid)
+            {
+                SetViewBagCourseOptions(customisationId, model.RefreshToCustomisationId);
+                return View("EditAutoRefreshOptions", model);
+            }
+
+            var data = TempData.Peek<EditLearningPathwayDefaultsData>()!;
+
+            var autoRefreshMonths =
+                model.AutoRefreshMonths == null ? 0 : int.Parse(model.AutoRefreshMonths);
+
+            var completeWithinMonthsInt =
+                data.LearningPathwayDefaultsModel.CompleteWithinMonths == null
+                    ? 0
+                    : int.Parse(data.LearningPathwayDefaultsModel.CompleteWithinMonths);
+
+            var validityMonthsInt =
+                data.LearningPathwayDefaultsModel.ValidityMonths == null
+                    ? 0
+                    : int.Parse(data.LearningPathwayDefaultsModel.ValidityMonths);

```

Comment:
Should be possible to pull these out into private methods to be used here and in the LearningDefaults post.